### PR TITLE
Feature: Piwik PRO analytics from Digital Analytic Platform Amsterdam

### PR DESCRIPTION
**Pull request by RubenSibon**

### CODE REVIEW

**Comment by RubenSibon** [comment]
D'oh, ik had de DAP URl moeten gebruiken ipv de eigen URL. 😅

File: src/shared/hooks/useAnalytics.ts [hunk 1]
```diff
@@ -1,33 +1,25 @@
...
-import PiwikTracker from '@amsterdam/piwik-tracker'
-import { UserOptions } from '@amsterdam/piwik-tracker/lib/types'
-import { usePiwik } from '@amsterdam/piwik-tracker-react'
+import PiwikPro, { PageViews } from '@piwikpro/react-piwik-pro'
 
-const rootUrl = process.env.REACT_APP_SELF_ROOT
-const siteId = process.env.REACT_APP_PIWIK_SITE_ID ?? ''
-const hasSiteId = !!process.env.REACT_APP_PIWIK_SITE_ID
+const dapUrl = 'https://dap.amsterdam.nl'
+const hasPiwikSiteId = !!process.env.REACT_APP_PIWIK_SITE_ID
```
Comment (TerryvanWalen) [inline src/shared/hooks/useAnalytics.ts:5]: - Why the `?? ''`? Seems unnecessary
- Might it be clearer if we did: `hasPiwikSiteId = !!piwikSiteId`

File: src/shared/hooks/useAnalytics.ts [hunk 2]
```diff
@@ -1,33 +1,25 @@
...
+const piwikSiteId = process.env.REACT_APP_PIWIK_SITE_ID ?? ''
 
-let PiwikInstance: PiwikTracker
+let PiwikInstance = false
 
-const PiwikTrackerConfig: UserOptions = {
-  urlBase: rootUrl,
-  siteId,
+function createPiwikInstance(isEnabled: boolean = true) {
+  if (isEnabled && hasPiwikSiteId && dapUrl && !PiwikInstance) {
```
Comment (TerryvanWalen) [inline src/shared/hooks/useAnalytics.ts:11]: If we enable analytics but don't have the necessary details like `piwikiSiteId` should we allow that? Should we not error or notify devs?

File: src/shared/hooks/useAnalytics.ts [hunk 3]
```diff
@@ -36,12 +28,13 @@ function useAnalytics() {
...
     },
-    [prevLocation, trackPageView]
+    [prevLocation]
   )
 
-  return { createPiwikInstance, trackPageVisit }
+  return { trackPageVisit }
 }
 
+export { piwikSiteId, hasPiwikSiteId, createPiwikInstance }
```
Comment (TerryvanWalen) [inline src/shared/hooks/useAnalytics.ts:39]: Why do we need to export these values? Especially the `piwikSiteId`? I can't think of a reason another part of the code would need this

**Review by TerryvanWalen — CHANGES_REQUESTED** [review]

File: src/shared/hooks/useAnalytics.ts (hunk 2 above)
Comment (RubenSibon) [reply src/shared/hooks/useAnalytics.ts]: Yeah, good point. I'll change this.

Which reminds me that this site does not have monitoring such as Sentry. Would be a good thing to add.

File: src/shared/hooks/useAnalytics.ts (hunk 3 above)
Comment (RubenSibon) [reply src/shared/hooks/useAnalytics.ts]: Good point. Removed.

File: src/shared/hooks/useAnalytics.ts (hunk 1 above)
Comment (TerryvanWalen) [reply src/shared/hooks/useAnalytics.ts]: Shouldn't this be from an env?
const dapUrl = 'https://dap.amsterdam.nl'

Comment (RubenSibon) [reply src/shared/hooks/useAnalytics.ts]: Aangepast

**Review by TerryvanWalen — APPROVED** [review]

**Review by TerryvanWalen — APPROVED** [review]

**Pull request merged**